server/command.go

```diff
@@ -481,6 +472,103 @@ func (p *Plugin) subscribeCommand(parameters []string, channelID string, config
 			fullPath,
 		)
 	}
-
 	return fmt.Sprintf("Successfully subscribed to %s.%s", fullPath, hookStatusMessage)
 }
+
+// subscribeCommand proccess the /gitlab subscribe command.
+// It returns a message and handles all errors my including helpful information in the message
+func (p *Plugin) subscribeCommand(parameters []string, channelID string, config *configuration, info *gitlab.GitlabUserInfo) string {
+	if len(parameters) == 0 {
+		return invalidSubscribeSubCommand
+	}
+
+	subcommand := parameters[0]
+
+	switch subcommand {
+	case "list":
+		return p.subscribeListCommand(channelID)
+	case "add":
+		features := "merges,issues,tag"
+		if len(parameters) > 2 {
+			features = strings.Join(parameters[2:], " ")
+		}
+		// Resolve namespace and project name
+		fullPath := normalizePath(parameters[1], config.GitlabURL)
+
+		return p.subscribeAddCommand(info, config, fullPath, channelID, features)
+	default:
+		return invalidSubscribeSubCommand
+	}
+
+}
+
+func getAutocompleteData() *model.AutocompleteData {
+	gitlabCommand := model.NewAutocompleteData("gitlab", "[command]", "Available commands: connect, disconnect, todo, subscribe, unsubscribe, me, settings, webhook")
+
+	connect := model.NewAutocompleteData("connect", "", "Connect your GitLab account")
+	gitlabCommand.AddCommand(connect)
+
+	disconnect := model.NewAutocompleteData("disconnect", "", "disconnect your GitLab account")
+	gitlabCommand.AddCommand(disconnect)
+
+	todo := model.NewAutocompleteData("todo", "", "Get a list of unread messages and merge requests awaiting your review")
+	gitlabCommand.AddCommand(todo)
+
+	subscribe := model.NewAutocompleteData("subscribe", "[command]", "Available commands: Add, list")
```

Comment:
Maybe `subscriptions` makes more sense here, as `/gitlab subscriptions add` sounds more like a real sentence then `gitlab subscribe add`. Or `subscription`?

Reply:
You've convinced me, ```subscriptions``` better represents that this is a subcommand, not an action. Especially when adding ```/gitlab subscriptions delete```. Might want also want to consider whether we like plural or singular subcommands. Most CLI's prefer singular, but plural might make the commands closer to natural language. 

My only other concern is that the GitLab and GitHub subcommands are consistent.

Reply:
There is an open PR to add autocomplete to the GitHub plugin: https://github.com/mattermost/mattermost-plugin-github/pull/309. The changes here should be applied there.

Reply:
@aaronrothschild @iomodo What do you think about the `/gitlab subscriptions [delete|add|list]` idea?

Reply:
I like the `subscriptions` subcommand. But probably this should be an alias not causing a "breaking" change. I'm 0/5 on this, but we could leave `/gitlab subscribe ..` as a working command and leave it out of the autocomplete.  Do our customers use some kind of automation that we might be breaking otherwise? @aaronrothschild ?

Reply:
Agreed, let's head in the right direction (Subscriptions) but not break the past (Subscribe).